torch_xla/experimental/xla_sharding.py

```diff
@@ -70,6 +71,128 @@ def get_logical_mesh(self):
     return self.device_ids.reshape(self.mesh_shape)
 
 
+class HybridMesh(Mesh):
+  ici_mesh_shape: Tuple[int, ...]
+  dcn_mesh_shape: Tuple[int, ...]
+
+  def __init__(self,
+               device_ids: Union[np.ndarray, List],
+               ici_mesh_shape: Tuple[int, ...],
+               dcn_mesh_shape: Tuple[int, ...] = None,
+               axis_names: Tuple[str, ...] = None):
+    if dcn_mesh_shape == None:
+      dcn_mesh_shape = tuple([1] * len(ici_mesh_shape))
+    mesh_shape = tuple([x * y for x, y in zip(ici_mesh_shape, dcn_mesh_shape)])
+    assert len(ici_mesh_shape) == len(dcn_mesh_shape)
+    super().__init__(device_ids, mesh_shape, axis_names)
+    self.device_attributes = pjrt.global_device_attributes()
+    if 'slice_index' in self.device_attributes[0] and np.prod(
+        dcn_mesh_shape) == 1:
+      raise ValueError('Provide dcn_mesh_shape to create a mesh for multislice')
+    self.ici_mesh_shape = ici_mesh_shape
+    self.dcn_mesh_shape = dcn_mesh_shape
+    if np.prod(dcn_mesh_shape) > 1:  # multislice
```

Comment:
What if we don't have 'slice_index' in this case? I guess you need to expand the condition on the Line 89.

Reply:
added an extra check